src/main.rs

```diff
@@ -287,28 +287,29 @@ fn main() {
                             Ok(val) => {
                                 trace!("file changed {:?}", val);
                                 if let Some(path) = val.path {
-                                    if path.is_absolute() {
-                                        // get where process was run from
-                                        let cwd = std::env::current_dir().unwrap_or(PathBuf::new());
-                                        // strip absolute path
-                                        let rel_path = path.strip_prefix(&cwd).unwrap_or(&cwd);
-
-                                        let path_starts_with_build =
-                                            &config.ignore.iter().any(|pattern| {
-                                                Pattern::matches_path(pattern, rel_path)
-                                            });
-                                        if !path_starts_with_build {
-                                            build(&config);
+                                    // get where process was run from
+                                    let cwd = std::env::current_dir().unwrap_or(PathBuf::new());
+
+                                    // the final goal is to have a relative path
+                                    // without corner-case prefixes like "./"
+                                    let abs_path = match path.is_absolute() {
+                                        true => path.clone(),
```

Comment:
`&path` is used in `unwrap_or` of `rel_path`, so I have to `path.clone()` here to have same datatype to `true` and `false branches.